### ISSUE TITLE
how to make loss visualization

### ISSUE THREAD

**Issue by 18842505953** [KaiyangZhou/deep-person-reid#148]
如何使训练损失loss通过图的形式呈现

**Comment by KaiyangZhou** [comment]
Try the following script

```
import argparse
import glob
import os
import os.path as osp
import re
import numpy as np
from matplotlib import pyplot as plt
import sys


parser = argparse.ArgumentParser(description='Parse log and draw curve for a specific keyword')
parser.add_argument('--file', type=str, default='')
parser.add_argument('--save-dir', type=str, default='')
parser.add_argument('--save-name', type=str, default='')
parser.add_argument('--kw', type=str, default='Loss')
args = parser.parse_args()

args.file = osp.expanduser(args.file)
if not osp.isfile(args.file):
    raise ValueError('File not found at {}'.format(args.file))
if not args.save_dir:
    args.save_dir = osp.dirname(args.file)
    print('Argument save_dir is not specified, so will be set to {}'.format(args.save_dir))
else:
    args.save_dir = osp.expanduser(args.save_dir)
if not args.save_name:
    args.save_name = args.kw + '.jpg'
    print('Argument save_name is not specified, so will be set to {}'.format(args.save_name))

regex = re.compile('{} ([-?.\d]+) \(([-?.\d]+)\)'.format(args.kw))
losses = []

print('Parsing log ...')

with open(args.file, 'r') as f:
    lines = f.readlines()
    for line in lines:
        match = regex.search(line)
        if match:
            loss = float(match.group(2))
            losses.append(loss)

if len(losses) == 0:
    raise RuntimeError('No values are detected, please check if {} is a valid keyword'.format(args.kw))

print('Plotting graph ...')
fig, ax = plt.subplots()
ax.plot(losses)
ax.set_xlabel('Step')
ax.set_title('{}'.format(args.kw))
save_path = osp.join(args.save_dir, args.save_name)
fig.savefig(save_path, bbox_inches='tight')
print('Graph is saved to {}'.format(save_path))

print('Done')
```

Say the log is saved under "results/train.log" and you wanna visualize the loss curve, which is saved under the keyword "Loss", do `python this_script.py --file results/train.log --kw Loss`.

**Comment by 18842505953** [comment]
非常感谢

**Comment by KaiyangZhou** [comment]
[tensorboardX](https://github.com/lanpa/tensorboardX) is also a good choice, I will consider adding this to the code in the near future.

I am currently overwhelmed by deadlines so several new features for this repo are postponed.
